src/parsers/mimic.py

```diff
@@ -157,68 +165,86 @@ def generate_lab_vect(self):
 
-        return t_c_l
+        return merged_chart_lab_events
 
-    def load_lab(self, h_med_adm1, h_med_adm2, use_lab_vect=True):
+    def load_lab(self, h_med_adm1, load_from_raw=True):
         """
         Load lab test data from LABEVENTS and CHARTEVENTS tables
         """
-        if use_lab_vect and os.path.exists(os.path.join(self.data, "mimiciii/1.4/preprocessed", "lab_patient_data_mimic_extract_2.csv")):
-            labs = pd.read_csv(os.path.join(self.data, "mimiciii/1.4/preprocessed", "lab_patient_data_mimic_extract_2.csv")) 
+        
+        if not load_from_raw and os.path.exists(os.path.join(self.data, constants.MIMIC_III_PREPROCESSED_PATH, "lab_patient_data_mimic_extract_2.csv")):
+            labs = pd.read_csv(os.path.join(self.data, constants.MIMIC_III_PREPROCESSED_PATH, "lab_patient_data_mimic_extract_2.csv")) 
         else:
-            self.generate_lab_vect()
-        labs = labs[labs["AGE"]<100]
+            labs = self.generate_lab_vect()
+
         labs = labs.drop(columns=["Unnamed: 0"])
-        labs = labs[labs.HADM_ID.isin(h_med_adm1+h_med_adm2)]
+        labs = labs[labs.HADM_ID.isin(h_med_adm1)]
+        
+        # Stratification
         labs = labs[labs["AGE"]>=self.age_b]
         labs = labs[labs["AGE"]<=self.age_a]
         labs = labs[labs["GENDER"]==self.gender] if self.gender != "MF" else labs
         labs = labs[labs["ETHNICITY"]==self.ethnicity]
+        
+        # Adding other column variables
         labs["CHARTTIME"] = pd.to_datetime(labs["CHARTTIME"])
         labs["ADMITTIME"] = pd.to_datetime(labs["ADMITTIME"])
         labs["LabTimeFromAdmit"] = labs["CHARTTIME"]-labs["ADMITTIME"]
         labs["hours_in"] = labs["LabTimeFromAdmit"].dt.total_seconds()/3600
+        
         return labs
 
-    def parse(self, use_pairs=True):
-        """
-        Loading medication and lab test. Performing basic preprocessing on data.
+    def parse(self, use_pairs=True, load_from_raw=True):
+        """Loading medication and lab test. Performing basic preprocessing on data.
+        
+        Args:
+            use_pairs (bool, optional): _description_. Defaults to True.
+            load_from_raw (bool, optional): Load preprocessed med files from raw MIMIC tables instead of using the previosu prepocessed files. Defaults to True.
+
+        Returns:
+            pd.DataFrame: 1st medication data
+            pd.DataFrame: 2nd medication data
+            pd.DataFrame: Lab tests data
         """
-        med1, hadm1 = self.load_med1()
-        med2, hadm2 = self.load_med2()
-        labs = self.load_lab(hadm1, hadm2)
+        med_preprocessed_path = os.path.join(self.data, constants.MIMIC_III_PREPROCESSED_PATH, constants.MIMIC_III_MED_PREPROCESSED_FILE_PATH)
+        med_preprocessed = pd.read_csv(med_preprocessed_path) if not load_from_raw and os.path.exists(med_preprocessed_path) else self.generate_med_data()
+        med_preprocessed = self.stratify_med_data(med_preprocessed)
```

Comment:
`generate_med_data()` does not produce the features AGE, and GENDER, and therefore this function failed.